### PR TITLE
Fix axeprep function to not look for fringe column if backgr is False

### PR DESCRIPTION
**Pull request by rosteen**
Previously, the `axeprep` function in `axetasks.py` would look for a `fringe` row in the `axe_inputs` even if the inputs indicated that there was no background provided (note that `fringe` in `axetasks` and `backims` in `inputchecks`/`axeinputs` are the same thing - a consistency pass is out of scope here). Now the code only looks for a `fringe` column if the `backgr` variable is set to True.

### CODE REVIEW

**Review by larrybradley — COMMENTED** [review]
LGTM.  Thanks, @rosteen.

I noticed that https://github.com/spacetelescope/hstaxe/blob/main/hstaxe/axesrc/inputchecks.py#L162 should be updated to be `if backims is None:`

**Comment by rosteen** [comment]
I figured that would be ok to leave as `not backims` because that would also catch user input of a blank string. I can change it to `if backims is not None and backims != ''` if being more explicit would be preferable.

**Comment by larrybradley** [comment]
Ah, ok.  Yeah, since `backims` can be `None` or `''`, I'd add the more explicit check.

**Review by larrybradley — APPROVED** [review]
Thanks, @rosteen!

**Comment by dsom-stsci** [comment]
Hi @rosteen @larrybradley: great, thanks! I think the flexibility of having 'backims' allowed to take None or ' ' in case of no background image is good. Previously we have seen at least one case where user specified 'None' threw up the key error. Also, can we put a note in the documentation that backims can take None or ' ' on case of no background image?

**Comment by rosteen** [comment]
@dsom-stsci This seems like it should be a silly question, but given the state of the HSTaXe documentation/my lack of familiarity with it - where would be a good place in the documentation (and which documentation) to put this note? 

Where are the readthedocs-style docs generated from this package even hosted? I know hstaxe.readthedocs.io doesn't exist, maybe we should make a ticket for making the docs available there. The user manual available from https://www.stsci.edu/scientific-community/software/axe appears similar, but not identical, to what I get when I build the `docs` folder from this package.

**Comment by dsom-stsci** [comment]
@rosteen , perhaps here? https://github.com/spacetelescope/hstaxe/blob/2186b552ed8c65d7268bc5d6bb41efe04e3dc5c2/hstaxe/axesrc/axetasks.py#L148

I am not entirely sure but this seems to be the best place to add this. What do you think @st-apidgeon ?

File: hstaxe/axesrc/inputchecks.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def check_axeprep(self, backgr, backims):
 
         # check for background subtraction
         if backgr:
-            if not backims:
+            if backims is None or backims == "":
```
Comment (duytnguyendtn) [inline hstaxe/axesrc/inputchecks.py:162]:
```suggestion
            if backims in (None, "")
```

**Review by duytnguyendtn — APPROVED** [review]
As far as my limited knowledge goes, this looks fine to me! One small nitpick if you're interested

**Pull request merged**